.github/workflows/backwards_compatibility_marqo_execution.yml

```diff
@@ -0,0 +1,167 @@
+        required: true
+      to_version:
+        description: 'Target Marqo version'
+        required: true
+      to_version_digest:
+        description: 'Target Marqo version digest'
+        required: true
+      from_image:
+        description: 'Source Marqo image (optional)'
+        required: false
+      to_image:
+        description: 'Target Marqo image (optional)'
+        required: false
+
+jobs:
+  Start-Runner:
+    permissions: write-all
+    name: Start self-hosted EC2 runner
+    runs-on: ubuntu-latest
+    outputs:
+      label: ${{ steps.start-ec2-runner.outputs.label }}
+      ec2-instance-id: ${{ steps.start-ec2-runner.outputs.ec2-instance-id }}
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.MARQO_WORKFLOW_TESTS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.MARQO_WORKFLOW_TESTS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+      - name: Start EC2 runner
+        id: start-ec2-runner
+        uses: machulav/ec2-github-runner@v2
+        with:
+          mode: start
+          github-token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          ec2-image-id: ${{ secrets.MARQO_CPU_AMD64_TESTS_INSTANCE_AMI }}
+          ec2-instance-type: m6i.xlarge
+          subnet-id: ${{ secrets.MARQO_WORKFLOW_TESTS_SUBNET_ID }}
+          security-group-id: ${{ secrets.MARQO_WORKFLOW_TESTS_SECURITY_GROUP_ID }}
+          aws-resource-tags: > # optional, requires additional permissions
+            [
+              {"Key": "Name", "Value": "marqo-compatibility-test-runner-${{ github.run_id }}"},
+              {"Key": "GitHubRepo", "Value": "${{ github.repository }}"},
+              {"Key": "WorkflowName", "Value": "${{ github.workflow }}"},
+              {"Key": "WorkflowRunId", "Value": "${{ github.run_id }}"},
+              {"Key": "WorlflowURL", "Value": "${{ github.event.repository.html_url }}/actions/runs/${{ github.run_id }}"},
+              {"Key": "PoloRole", "Value": "testing"}
+            ]
+            
+  backwards_compatibility:
+    # This job runs on the newly created runner
+    runs-on: ${{ needs.start-runner.outputs.label }}
+    needs: Start-Runner
+    steps:
+      # Step to check out the Marqo repository
+      - name: checkout marqo repo
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ github.repository }} #Check out the repository that contains this action since the tests exist in the same repository
+          fetch-depth: 0
+
+      # Step to set up Python 3.9
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.9"
+          cache: "pip"
+
+      # Step to install pip-tools
+      - name: Install pip-tools
+        run: |
+          python -m pip install pip-tools
+
+      - name: Debug
+        run: |
+          pwd && ls -laR && echo $GITHUB_WORKSPACE
```

Comment:
What is this doing?

Reply:
I was using this for debugging - I'll delete it